docs/02-app/02-api-reference/02-file-conventions/route-segment-config.mdx

```diff
@@ -197,6 +200,20 @@ Support for `preferredRegion`, and regions supported, is dependent on your deplo
 > - If a `preferredRegion` is not specified, it will inherit the option of the nearest parent layout.
 > - The root layout defaults to `all` regions.
 
+### `maxDuration`
+
+```tsx filename="layout.tsx / page.tsx / route.ts" switcher
+export const maxDuration = 5
+```
+
+```js filename="layout.js / page.js / route.js" switcher
+export const maxDuration = 5
+```
+
+> **Good to know**:
+>
+> - If a `maxDuration` is not specified, the default is the [function timeout limit](https://vercel.com/docs/concepts/limits/overview#general-limits) of your Vercel plan.
```

Comment:
This also works self-hosted, although likely not as interesting, but we'll need to make that more clear.

Reply:
We need a sentence on what this is and why it matters. Then, if you're using Vercel, here's what it does there, for example.